### ISSUE TITLE
关于batchnorm层改变神经元模电压的问题/How does the batchnorm layer change the neuron mode voltage?

### ISSUE THREAD

**Issue by jamieyang66** [fangwei123456/spikingjelly#294]
我尝试使用conv-batchnorm-IFNode层。我发现，假设从conv层输出的x为0，那么经过batchnorm层后，会有一个bias，这个bias在最初的时候会改变IFNode的电压到v=bias。但是如果下一时刻从conv层输出的x仍然为0，那么经过batchnorm层后，还是会有一个bias，但是这个bias似乎不会再次改变IFNode的电压到2bias。这是我通过监控器观察到的情况，不知道是否正确。我希望能明确batchnorm的电压改变逻辑。
I tried using the conv batchnorm IFNode layer. I found that if the x output from the conv layer is 0, there will be a bias after the batchnorm layer. This bias will initially change the voltage of the IFNode to v=bias. However, if the x output from the conv layer is still 0 at the next time, there will still be a bias after the batchnorm layer, but this bias does not seem to change the voltage of the IFNode again to 2bias. This is what I observed through the monitor. I don't know whether it is correct. I hope to be able to define the voltage change logic of batchnorm.

**Comment by fangwei123456** [comment]
提供一下最小复现的代码？

**Comment by jamieyang66** [comment]
[mnist_code.zip](https://github.com/fangwei123456/spikingjelly/files/9909794/mnist_code.zip)
代码的粘贴似乎有问题，所以我打包成了一个压缩包文件，包括一个mnist_train代码和一个mnist_test代码
在mnist_train我训练了一个小的SNN网络
在mnist_test中我是用monitor来监控IFNode的输入电压和神经元内部电压，发现许多神经元在不同的timestep上都没有电压的变化，保持着对应channel的bias电压没有变化，即使后面持续输入bias电压，也没有递增

**Comment by jamieyang66** [comment]
非常抱歉这是我第一次用issue，不太熟悉粘贴代码，我尝试了一下粘贴代码，但是Preview以后非常难看

**Comment by fangwei123456** [comment]
似乎是监视器的用法错误。
![image](https://user-images.githubusercontent.com/38352109/199234577-9f127159-59e9-4504-b717-71baa9493a9a.png)

**Comment by fangwei123456** [comment]
`print("add:",input_monitor.records[1])`

永远都是打印记录的第一个数据，因此完全相同

**Comment by jamieyang66** [comment]
这是我用以下代码监控的某一行，可以看到电压没有任何变化。
<img width="503" alt="example" src="https://user-images.githubusercontent.com/117155961/199235745-a07b4bc7-02e4-455c-abde-465f35386e0c.PNG">

    input_monitor = monitor.InputMonitor(net, neuron.IFNode)
    mtor = monitor.AttributeMonitor('v', False, net, instance=neuron.IFNode)      
    with torch.no_grad():
        for img, label in test_data_loader:
            inum=inum+1
            img = img.to('cuda:0')
            label = label.to('cuda:0')
            label_onehot = F.one_hot(label, 10).float()
            out_fr = 0.
            for t in range(10):
                encoded_img = encoder(img)
                if t>=1:
                  print("step:",t)
                  print("add:",input_monitor.records[1])
                  print("result:",mtor.records[1])
                out_fr += net(encoded_img)

**Comment by jamieyang66** [comment]
我看了，确实是您说的这样，我以为只记录当前时刻的，没想到都记录了，谢谢！
